crates/swc_ecma_lints/src/config.rs

```diff
@@ -1,44 +1,61 @@
-use crate::rules::no_console::NoConsoleConfig;
-use serde::{Deserialize, Serialize};
-use std::fmt::Debug;
-
-#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
-#[serde(rename_all = "lowercase")]
-pub enum LintRuleReaction {
-    Off,
-    Warning,
-    Error,
-}
+#[cfg(feature = "non_critical_lints")]
+#[path = ""]
+mod non_critical_lints {
+    use crate::rules::non_critical_lints::no_console::NoConsoleConfig;
+    use serde::{Deserialize, Serialize};
+    use std::fmt::Debug;
 
-impl Default for LintRuleReaction {
-    fn default() -> Self {
-        Self::Off
+    #[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+    #[serde(rename_all = "lowercase")]
+    pub enum LintRuleReaction {
+        Off,
+        Warning,
+        Error,
     }
-}
 
-#[derive(Debug, Clone, Default, Serialize, Deserialize)]
-pub struct RuleConfig<T: Debug + Clone + Serialize + Default>(
-    #[serde(default)] LintRuleReaction,
-    #[serde(default)] T,
-);
+    impl Default for LintRuleReaction {
+        fn default() -> Self {
+            Self::Off
+        }
+    }
+
+    #[derive(Debug, Clone, Default, Serialize, Deserialize)]
+    pub struct RuleConfig<T: Debug + Clone + Serialize + Default>(
+        #[serde(default)] LintRuleReaction,
+        #[serde(default)] T,
+    );
+
+    impl<T: Debug + Clone + Serialize + Default> RuleConfig<T> {
+        pub(crate) fn get_rule_reaction(&self) -> &LintRuleReaction {
+            &self.0
+        }
 
-impl<T: Debug + Clone + Serialize + Default> RuleConfig<T> {
-    pub(crate) fn get_rule_reaction(&self) -> &LintRuleReaction {
-        &self.0
+        pub(crate) fn get_rule_config(&self) -> &T {
+            &self.1
+        }
     }
 
-    pub(crate) fn get_rule_config(&self) -> &T {
-        &self.1
+    #[derive(Debug, Clone, Default, Serialize, Deserialize)]
+    #[non_exhaustive]
+    #[serde(rename_all = "camelCase")]
+    pub struct LintConfig {
+        #[serde(default)]
+        pub no_console: RuleConfig<NoConsoleConfig>,
+
+        #[serde(default)]
+        pub no_alert: RuleConfig<()>,
+
+        #[serde(default)]
+        pub no_debugger: RuleConfig<()>,
     }
 }
 
-#[derive(Debug, Clone, Default, Serialize, Deserialize)]
-#[non_exhaustive]
-#[serde(rename_all = "camelCase")]
-pub struct LintConfig {
-    #[serde(default)]
-    pub no_console: RuleConfig<NoConsoleConfig>,
-
-    #[serde(default)]
-    pub no_debugger: RuleConfig<()>,
+#[cfg(not(feature = "non_critical_lints"))]
```

Comment:
This will make these lints removed when the featrure is enabled.
Can you revert commits related to feature gates?

Reply:
A bit didn't understand you. Why lints will be removed if features is enabled? Here is `cfg(not(feature...`

Reply:
Because `#[cfg(not(feature = "non_critical_lints"))]` means this code (under `cfg`) should be built only if `not(feature = "non_critical_lints")`.
It means if we enable `non_critical_lints `, it's `not(true)`, so it's removed.

Reply:
then if we're not enable non_critical_lints then it's not(false) -> true.

if it is then it's expected.

here is I include all non critical lints if feature flag provided https://github.com/swc-project/swc/pull/3394/files/aade1992b77528596ce47ab63fa45ecf2e4374e3#diff-804fa350bc6183d7f8eb97b783ba6090005850a1f7a3c176f62b572871ff558dR1

and here is not https://github.com/swc-project/swc/pull/3394/files/aade1992b77528596ce47ab63fa45ecf2e4374e3#diff-804fa350bc6183d7f8eb97b783ba6090005850a1f7a3c176f62b572871ff558dR53

Reply:
Here I provide empty LintConfig struct if `non_critical_lints` feature flag isn't provided

Reply:
Possibly here is should be used another name

https://github.com/swc-project/swc/pull/3394/files/aade1992b77528596ce47ab63fa45ecf2e4374e3#diff-804fa350bc6183d7f8eb97b783ba6090005850a1f7a3c176f62b572871ff558dR54